### PR TITLE
[lineage] Add system source table lineage meta table

### LINKED ISSUE #2197: [Feature] Add system table for source table lineage meta
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Add system table for source lineage table meta, subtask of #684 

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by FangYongs**
### Purpose

Linked issue: close #2197 

This pr aims to add system table `SourceTableLineageTable` and users can query for source table linage by sql, for example:
SELECT * FROM sys.source_table_lineage;

### Tests

1. Added source table lineage in test `FlinkLineageITCase#testTableLineage`

### API and Format

no

### Documentation

no

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/table/system/SystemTableLoader.java [hunk 1]
```diff
@@ -75,12 +78,18 @@ public static Table loadGlobal(
             String tableName,
             FileIO fileIO,
             Supplier<Map<String, Map<String, Path>>> allTablePaths,
-            Map<String, String> catalogOptions) {
+            Map<String, String> catalogOptions,
+            LineageMetaFactory lineageMetaFactory) {
```
Comment (schnappi17) [inline paimon-core/src/main/java/org/apache/paimon/table/system/SystemTableLoader.java:82]: @Nullable

File: paimon-core/src/main/java/org/apache/paimon/table/system/SourceTableLineageTable.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        return new RowType(fields);
+    }
+
+    @Override
+    public List<String> primaryKeys() {
+        return Arrays.asList("database_name", "table_name", "job_name");
+    }
+
+    @Override
+    public Table copy(Map<String, String> dynamicOptions) {
```
Comment (schnappi17) [inline paimon-core/src/main/java/org/apache/paimon/table/system/SourceTableLineageTable.java:127]: The dynamicOptions are ignored here, is it as expected?

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkLineageITCase.java [hunk 3]
```diff
@@ -72,43 +78,65 @@ public void testTableLineage() {
                 .hasRootCauseMessage("Method saveSinkTableLineage is not supported");
 
         tEnv.getConfig().getConfiguration().set(PipelineOptions.NAME, "select_t_job");
-        assertThatThrownBy(() -> tEnv.executeSql("SELECT * FROM T").collect().close())
-                .hasRootCauseInstanceOf(UnsupportedOperationException.class)
-                .hasRootCauseMessage("Method saveSourceTableLineage is not supported");
+        tEnv.executeSql("SELECT * FROM T").collect().close();
+        assertThat(jobSourceTableLineages.isEmpty()).isFalse();
```
Comment (schnappi17) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkLineageITCase.java:82]: isNotEmpty

File: paimon-core/src/main/java/org/apache/paimon/table/system/SourceTableLineageTable.java (hunk 2 above)
Comment (FangYongs) [reply paimon-core/src/main/java/org/apache/paimon/table/system/SourceTableLineageTable.java:127]: It's useless for lineage tables.

**Comment by liming30** [comment]
+1

**Review by liming30 — COMMENTED** [review]
+1

**Pull request merged**